presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -334,6 +338,22 @@ public void checkCanDropSchema(SystemSecurityContext context, CatalogSchemaName
         if (!canAccessCatalog(context.getIdentity(), schema.getCatalogName(), ALL)) {
             denyDropSchema(schema.toString());
         }
+
+        if (schemaRules.isPresent() && !isSchemaOwner(context, schema.getSchemaName())) {
+            denyDropSchema(schema.getSchemaName());
+        }
+    }
+
+    private boolean isSchemaOwner(SystemSecurityContext context, String schemaName)
+    {
```

Comment:
`if (schemaRules.isEmpty()) { return true; }`

Then you don't need to have `schemaRules.isPresent() && ` in all the cases above.

Reply:
@kokosing - I have made the changes and pushed